# Settings for ros_control control loop
hardware_control_loop:
   loop_hz: 100

# Settings for ros_control hardware interface
hardware_interface:
   joints:
     - shoulder_joint
     - upperArm_joint
     - foreArm_joint
     - wrist1_joint
     - wrist2_joint
     - wrist3_joint

# Publish all joint states ----------------------------------
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 100

# Publish wrench ----------------------------------
#force_torque_sensor_controller:
#   type: force_torque_sensor_controller/ForceTorqueSensorController
#   publish_rate: 100

# Joint Trajectory Controller - position based -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
pos_based_pos_traj_controller:
   type: position_controllers/JointTrajectoryController
   joints:
     - shoulder_joint
     - upperArm_joint
     - foreArm_joint
     - wrist1_joint
     - wrist2_joint
     - wrist3_joint
   constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_joint: {trajectory: 0.1, goal: 0.1}
      upperArm_joint: {trajectory: 0.1, goal: 0.1}
      foreArm_joint: {trajectory: 0.1, goal: 0.1}
      wrist1_joint: {trajectory: 0.1, goal: 0.1}
      wrist2_joint: {trajectory: 0.1, goal: 0.1}
      wrist3_joint: {trajectory: 0.1, goal: 0.1}
   stop_trajectory_duration: 0.5
   state_publish_rate:  100
   action_monitor_rate: 10
   gains:
      #!!These values have not been optimized!!
      shoulder_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
      upperArm_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
      foreArm_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
      wrist1_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
      wrist2_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
      wrist3_joint: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}

   # state_publish_rate:  50 # Defaults to 50
   # action_monitor_rate: 20 # Defaults to 20
   #hold_trajectory_duration: 0 # Defaults to 0.5
   
# Joint Trajectory Controller - velocity based -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
vel_based_pos_traj_controller:
   type: velocity_controllers/JointTrajectoryController
   joints:
     - shoulder_joint
     - upperArm_joint
     - foreArm_joint
     - wrist1_joint
     - wrist2_joint
     - wrist3_joint
   constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_joint: {trajectory: 0.1, goal: 0.1}
      upperArm_joint: {trajectory: 0.1, goal: 0.1}
      foreArm_joint: {trajectory: 0.1, goal: 0.1}
      wrist1_joint: {trajectory: 0.1, goal: 0.1}
      wrist2_joint: {trajectory: 0.1, goal: 0.1}
      wrist3_joint: {trajectory: 0.1, goal: 0.1}
   stop_trajectory_duration: 0.5
   state_publish_rate:  100
   action_monitor_rate: 10
   gains:
      #!!These values are useable, but maybe not optimal
      shoulder_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}
      upperArm_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}
      foreArm_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}
      wrist1_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}
      wrist2_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}
      wrist3_joint: {p: 1.2,  i: 0.0, d: 0.1, i_clamp: 1}

   # state_publish_rate:  50 # Defaults to 50
   # action_monitor_rate: 20 # Defaults to 20
   #hold_trajectory_duration: 0 # Defaults to 0.5

